### PR TITLE
Add a build lane for FreeBSD Mono.

### LINKED ISSUE #57487: Add FreeBSD to mono build in runtime.yml
I have PR https://github.com/dotnet/runtime/pull/57027 in progress with some fixes to allow the mono subset to be built under native FreeBSD as well as cross-built under Linux.
FreeBSD is currently enabled for CI build (Linux cross-build) under these sections:

- CoreCLR Product Build FreeBSD x64 release
- Libraries Build FreeBSD x64 Debug
- Installer Build and Test coreclr FreeBSD_x64 Release

At the moment, `eng/pipelines/runtime.yml` doesn't have FreeBSD under any of the mono build sections.

After https://github.com/dotnet/runtime/pull/57027 is merged, could the owners of `eng/pipelines/runtime.yml` please investigate adding FreeBSD to the relevant section/s.

### PR DESCRIPTION
**Pull request by directhex**
No test build/run, as there's no Helix queue for it.

Fixes: #57487

### CODE REVIEW

**Review by fanyang-mono — APPROVED** [review]

File: eng/pipelines/runtime-staging.yml [hunk 1]
```diff
@@ -99,6 +99,35 @@ jobs:
...
+      testScope: innerloop
+      nameSuffix: AllSubsets_Mono
+      buildArgs: -s mono+libs+host+packs -c $(_BuildConfig) /p:ArchiveTests=true /p:UseLiveBuiltDotNetHost=true
+      timeoutInMinutes: 120
+      condition: >-
+        or(
+          eq(dependencies.evaluate_paths.outputs['SetPathVars_libraries.containsChange'], true),
+          eq(dependencies.evaluate_paths.outputs['SetPathVars_mono.containsChange'], true),
+          eq(dependencies.evaluate_paths.outputs['SetPathVars_installer.containsChange'], true),
+          eq(variables['isRollingBuild'], true))
```
Comment (jkotas) [inline eng/pipelines/runtime-staging.yml:129]: Do we need to run this in the rolling build?

Comment (directhex) [reply eng/pipelines/runtime-staging.yml:129]: Whatever you prefer

File: eng/pipelines/runtime-staging.yml [hunk 2]
```diff
@@ -99,6 +99,34 @@ jobs:
...
+    buildConfig: Release
+    runtimeFlavor: mono
+    platforms:
+    - FreeBSD_x64
+    variables:
+      # map dependencies variables to local variables
+      - name: librariesContainsChange
+        value: $[ dependencies.evaluate_paths.outputs['SetPathVars_libraries.containsChange'] ]
+      - name: monoContainsChange
+        value: $[ dependencies.evaluate_paths.outputs['SetPathVars_mono.containsChange'] ]
```
Comment (jkotas) [inline eng/pipelines/runtime-staging.yml:118]:
```suggestion
```
These variables are not used anywhere. Do we need them?

File: eng/pipelines/runtime-staging.yml [hunk 3]
```diff
@@ -99,6 +99,34 @@ jobs:
...
+    jobParameters:
+      testScope: innerloop
+      nameSuffix: AllSubsets_Mono
+      buildArgs: -s mono+libs+host+packs -c $(_BuildConfig) /p:ArchiveTests=true /p:UseLiveBuiltDotNetHost=true
```
Comment (jkotas) [inline eng/pipelines/runtime-staging.yml:122]:
```suggestion
      buildArgs: -s mono+libs+host+packs -c $(_BuildConfig)
```
This is not doing any testing, so these args should not be needed.

**Pull request merged**